walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,78 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    //If no attendees are able to attend
+    if(request.getAttendees().isEmpty() && events.isEmpty()){
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    //If greater than lenght of day
+    if(request.getDuration() > TimeRange.WHOLE_DAY.duration()){
+        return Arrays.asList();
+    }
+
+    //return freetimes based on attendees
+    if(events.isEmpty() == false && request.getAttendees().isEmpty() == false){ //correct conditions?
+        ArrayList<TimeRange> bookedTimes = new ArrayList();
+        ArrayList<TimeRange> unbookedTimes = new ArrayList();
+        int earliestFreeTime = TimeRange.START_OF_DAY;
+        int latestEndTime = TimeRange.END_OF_DAY;
+        Event previousEvent = events.iterator().next();
+        for(Event event: events){
+            int startEvent = event.getWhen().start();
+            int endEvent = event.getWhen().end();
+            if(event.getWhen().overlaps(previousEvent.getWhen())){
+                TimeRange combined = combine(event, previousEvent);
+                bookedTimes.add(TimeRange.fromStartEnd(combined.start(), combined.end(), false));
+            }else{
+                bookedTimes.add(TimeRange.fromStartEnd(startEvent, endEvent, false));
+            }
+            previousEvent = event;
+        }
+
+        for(TimeRange bookedTime: bookedTimes){
+            if(earliestFreeTime < latestEndTime){
```

Comment:
See if you can reduce the nesting here, it's a lot to try to trace through this tangle of if/else statements

See some advice here:
https://drive.google.com/file/d/0B-KFDiTrQe9lLU9uVDZzZWNmRVU/view